[24기 김영현] [Mission 1~7 완료 / 추가 기능 구현 진행 중 🔥 ] Westagram 업데이트

dudgus5766 opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
Westagram login, main 페이지 업로드
		 
## 수정 사항들 자세한 내용
- login 페이지 작성 완료
- main 페이지 html, css 작성  완료
- main 페이지 댓글 이벤트 구현
- mission 1~7번 완료

## 기타 질문 및 특이 사항
- 로그인 페이지의 input의 placeholder를 span으로 바꿔 플로팅 기능 구현 중

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

joonsikyang commented on students/11th/yejikang/style/login.css:
```diff
@@ -0,0 +1,80 @@
+html,
```
html 태그에 스타일을 입힐 필요가 있을까요?

Kangyejiii replied:
수정했습니다! body { height: 100vh; } 로 body에만 스타일 적용했습니다.

joonsikyang commented on students/11th/yejikang/style/login.css:
```diff
@@ -0,0 +1,80 @@
+}
+
+/* 로그인폼 - 비밀번호 찾기 링크  */
+.lostPassword {
+  position: absolute;
+  bottom: 0px;
+}
```
position 속성 잘 활용하셨네요!

joonsikyang commented on students/11th/yejikang/js/login.js:
```diff
@@ -0,0 +1,20 @@
+const userId=document.querySelector('.user-id');
+const userPw = document.querySelector('.user-pw');
+
+function init(){
+    loginButton.disabled = true;
+}
+
+init();
```
```suggestion
(function init(){
    loginButton.disabled = true;
})()
```

위와 같이 즉시실행함수 형태로 바꿀 수 있겠네요. 아래 링크 참고해보세요 :)
https://developer.mozilla.org/ko/docs/Glossary/IIFE

Kangyejiii replied:
`()`를 이용해서 형태를 바꿔보았습니다 감사합니다!

joonsikyang commented on students/11th/yejikang/style/login.css:
```diff
@@ -0,0 +1,80 @@
+  font-weight: bold;
+}
+
+.loginBtn:hover:enabled {
+  cursor: pointer;
+}
+
+.loginBtn:enabled{
+  background-color: #0095F6; 
+}
```
css 활용 Good👍

joonsikyang commented on students/11th/yejikang/js/login.js:
```diff
@@ -0,0 +1,20 @@
+init();
+
+function setEnabledButton() {
+     loginButton.disabled =  (userId.value && userPw.value)? false : true ; 
```
삼항연산자 벌써 적용 잘해주셨네요!

Kangyejiii commented on students/11th/yejikang/js/login.js:
```diff
@@ -0,0 +1,20 @@
+const loginButton = document.querySelector('.loginBtn');
+const loginForm = document.querySelector('.loginForm');
+const userId=document.querySelector('.user-id');
+const userPw = document.querySelector('.user-pw');
```
넵!

saengmotmi commented on students/11th/yejikang/js/login.js:
```diff
@@ -0,0 +1,18 @@
+     loginButton.disabled =  (userId.value && userPw.value)? false : true ; 
+}
+
+loginForm.addEventListener('keyup',setEnabledButton);
+
+loginButton.addEventListener('click', function(){
+     console.log(userId.value,userPw.value);   
+});
```
```suggestion
loginForm.addEventListener('keyup',setEnabledButton);

loginButton.addEventListener('click', function() {
  console.log(userId.value, userPw.value);   
});
```
1. 띄어쓰기도 잘 해주셔야 코드 가독성이 좋아집니다
2. 테스트 끝난 console은 지워주세요
3. vscode에서 tab size 2칸으로 바꿔주세요

Kangyejiii replied:
넵 수정했습니다!

Kangyejiii replied:
네!

saengmotmi commented on students/11th/yejikang/js/main.js:
```diff
@@ -0,0 +1,36 @@
+const userId = "yejiii_95";  
+
+function addComment() {
+    if(inputComment.value.trim('').length===0) return;
```
```suggestion
    if(!inputComment.value.trim('')) return;
```
1. 자바스크립트에서 0은 falsy한 값입니다. 그 자체로 true, false 값을 갖는다는 뜻입니다.
2. 부정연산자(!)를 사용하면 불리언 값을 뒤집어줄 수 있습니다. if문의 조건식이 true가 되면 실행되므로 이 경우 위와 같이 입력하면 빈칸인 경우를 뜻하게 됩니다.

Kangyejiii replied:
수정했습니다.

Kangyejiii replied:
수정했습니다 감사합니다!

saengmotmi commented on students/11th/yejikang/js/main.js:
```diff
@@ -0,0 +1,36 @@
+    userIdSpan.classList.add('infoBold','commentId');
+    heartBtn.classList.add('commentHeartBtn', 'heart');
+
+    let id = document.createTextNode(userId);
```
id라는 변수를 let으로 선언한 특별한 이유가 있을까요?

Kangyejiii replied:
변하지 않는 id 값을 let으로 선언 할 이유가 없군요!! 수정하겠습니다. 감사합니다.

saengmotmi commented on students/11th/yejikang/js/main.js:
```diff
@@ -0,0 +1,36 @@
+commentBtn.addEventListener('click',addComment);
+
+inputComment.addEventListener('keyup',function(e){
+    if (e.keyCode == 13) {
```
가능하면 ===로 엄격한 비교를 하는 것이 좋습니다

Kangyejiii replied:
넵!

joonsikyang replied:
종택님은 엄격한 멘토입니다. 참고해주세요.

saengmotmi commented on students/11th/yejikang/login.html:
```diff
@@ -0,0 +1,32 @@
+    <main>
+        <div class="loginContainer">
+          <div class="logo">
+            <img src="img/logo_text.png" alt="instagram" />
```
alt 속성 잘 써주셨습니다. 다만 src 보다 앞으로 배치되는 것이 해당 이미지에 대한 설명이 더 잘 되겠네요.

saengmotmi commented on students/11th/yejikang/style/login.css:
```diff
@@ -0,0 +1,75 @@
+  width: 350px;
+  height: 380px;
+  padding: 0 0 10px 0;
+  border: 1px solid rgba(var(--b6a, 219, 219, 219), 1);
```
`--b6a` 변수가 어디에 선언되어 있나요?

참고 : https://developer.mozilla.org/ko/docs/Web/CSS/var

Kangyejiii replied:
앗..! 수정했습니다!

joonsikyang commented:
LGTM 🎯

dudgus5766 committed:
Add: 위스타그램 업데이트
- html, css, js 파일 업로드
- login 페이지 작성 완료
- main 페이지 html, css 작성  완료
- main 페이지 댓글 이벤트 구현

dudgus5766 committed:
Add : 댓글 추가 기능 구현

dudgus5766 committed:
Refactor : HTML&CSS 리팩토링
Add : 사진 추가

dudgus5766 committed:
🚧  검색창 구현 백업

dudgus5766 committed:
🚧 재귀 해야됨

dudgus5766 committed:
Add : 자동 완성 검색 기능 완성!

dudgus5766 committed:
Add : 검색창 클릭 시 검색창 생성
- 검색창 영역 외 부분을 클릭할 시에 검색창이 사라짐

dudgus5766 committed:
Add : 로그인 페이지 기능 구현
- ID input에 '@' 포함 조건 구현

dudgus5766 committed:
Add : 로그인 시 'alert'
- login button 조건으로 게시 버튼이 활성화 -> enter or 게시버튼 click 이벤트로 로그인 'alert'반환
